nfcd/src/main/jni/src/helper/MapInfo.cpp

```diff
@@ -13,8 +15,21 @@ bool MapInfo::create() {
     int rv = dl_iterate_phdr([] (struct dl_phdr_info *info, size_t, void *user_data) {
         auto *instance = (MapInfo *)user_data;
 
+        // Skip entries with null names to prevent crashes
+        if (!info->dlpi_name)
+            return 0;
+        
+        // Use memory address as unique identifier for entries with empty names
+        std::string libName;
+        if (strlen(info->dlpi_name) == 0) {
+            char addrStr[32];
+            snprintf(addrStr, sizeof(addrStr), "0x%lx", (unsigned long)info->dlpi_addr);
+            libName = addrStr;
+        } else
+            libName = info->dlpi_name;
```

Comment:
```suggestion
        auto libName = info->dlpi_name[0] ? info->dlpi_name : "<" + std::to_string(info->dlpi_addr) + ">";
```
A little less cumbersome